src/libstore/local-store.cc

```diff
@@ -158,7 +158,8 @@ void migrateCASchema(SQLite& db, Path schemaPath, AutoCloseFD& lockFd)
             txn.commit();
         }
 
-        writeFile(schemaPath, fmt("%d", nixCASchemaVersion));
+        writeFile(schemaPath, fmt("%d", nixCASchemaVersion), 0666, true);
+        syncParent(schemaPath);
```

Comment:
Shouldn't that be done automatically by `writeFile` when `sync` is set to true instead?

Reply:
That makes sense. Changed `writeFile` to call `syncParent`.